index.js

```diff
@@ -86,8 +86,23 @@ module.exports = function (str, opts) {
 		delete urlObj.search;
 	}
 
+	var queryParameters = queryString.parse(urlObj.search);
+
+	// remove query unwanted parameters
+	if (Array.isArray(opts.ignoredQueryParameters)) {
+		for (var key in queryParameters) {
+			if (!queryParameters.hasOwnProperty(key)) {
```

Comment:
Why `hasOwnProperty`? In the next version of `query-string`, the returned object won't have a prototype and thus this method will fail. I would go for this.

```js
Object.keys(queryParameters).forEach(function (key) {
    if(queryParameters[key] && testQueryParameter(key, opts.ignoredQueryParameters)) {
        delete queryParameters[key];
    }
});
```

I prefer `Object.keys()` above `for..in`, not sure why though. But apart from that, the same logic applies.